doc/release/0.17.0-notes.rst

```diff
@@ -29,6 +29,9 @@ Deprecated features
 Backwards incompatible changes
 ==============================
 
+The functions ``trim1`` and ``trimboth`` now sort elements before trimming.
+This is backwards incompatible but it made no sense to trim unsorted input.
```

Comment:
@juliantaylor yes, you can see `np.partition` being used as you mention in `trim_mean` because the ordering really doesn't matter either way. 

For `trim1` and `trimboth` there is a tradeoff:
**`np.sort`:** 
*  For small sized input (e.g. 1000 elements) the difference in performance between `np.sort` and `np.partition` seems negligible.
* Makes more sense in my opinion because this is not an in-place change (i.e. the result doesn't change the input array, but produce a new one) so it seems better to have the new result already sorted.

**`np.partition`:**
* For larger sized input (e.g. ~100,000 elements) `np.sort` time was around 2.5x that of `np.partition`

The tie braker was that having sorted output made more sense to me, but this is obviously an opinion. More feedback is appreciated.